### PR TITLE
Bugfix: Composer: text looses formatting after editing a markdown message or editing a reply message

### LINKED ISSUE #8602: RTE: text looses formatting after editing a markdown message
#### wrong formatting when editing a message:
1. enable markdown messages
2. send any message, e.g.
   ```md
   # test
   > foo
   ```
3. the message is sent correctly
4. edit the message - the message is rendered in the composer like in the chat (with big header and a cite) but the formatting marks are missing.
   in my case i just added a `/n/n´bar´` (i was not able to break/ exit the comment either)
5. submit message
6. the new edited message is (without the cite):
   > test <- not a header
   > 
   > foo <- not a cite
   > 
   > `bar` <- that works because i added the formatting marks
   
   if you add the formatting marks manually before submitting the edit, the message is sent correctly.

#### duplicate `in reply to …`:

when replying to any message and editing the reply the same bug occures:
1. markdown can be enabled but it is not required
2. send any message
3. reply to that message with any message
4. edit the repy - the message is rendered with the
   > > in reply to @…
   > \*initial message\*
   > 
   > \*reply message\*
   which is applied to the message after submitting the edit draft.
5. the edited message is:
   > > in reply to @…
   > > \*initial message\*
   > 
   > in reply to @…
   > \*initial message\*
   > 
   > \*reply message\*

there might be much more bugs when editing messages i haven't found.

#### What did you expect?

the message is rendered correctly

#### What happened instead?

i believe the problem is that the rendered message text is put into the composer instead of the raw message text so all formatting of the message gets lost when editing

### Your phone model

OnePlus 5T

### Operating system version

Android 10

### Application version and app store

Element version v1.6.5 from Google Play

### Homeserver

Synapse v1.89.0

### Will you send logs?

No

### Are you willing to provide a PR?

No

### LINKED ISSUE #8612: Reply quote is present when editing a message
### Steps to reproduce

1. DM Chat with someone
2. (Send a message)
3. Reply to a message
4. Edit the reply

#### What did you expect?

I expected that the content that appears in the edit box would only be the reply text.

#### What happened instead?

The edit box also contained the "From" and original text (see video)

### Your phone model

Google Pixel 6a

### Operating system version

Android 13 (TQ3A.230705.001)

### Application version and app store

Element 1.6.5 [40106052] (G-fd6a45a3), Rust SDK 0.6.0 (3fd55c0), Vodozemac 0.4.0 from Play Store

### Homeserver

Synapse 1.85.2

### Will you send logs?

No

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by Luka5W**
## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Reverted some changes made in #8377

## Motivation and context

Fixes #8602
Fixes #8612

## Screenshots / GIFs

The screenshots were made immediately after long-tapping on the message to edit and selecting edit:

|Before|After|
|-|-|
|Screenshot_20230902_013913|Screenshot_20230902_013050|

## Tests

See screenshots

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Pixel_3a_API_34_extension_level_7_x86_64

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: changelog.d/8638.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+User pills don't get lost at message editing anymore
```
Comment (pludi) [inline changelog.d/8638.bugfix:1]: `pills`, or `polls`?

Comment (Luka5W) [reply changelog.d/8638.bugfix:1]: well... i have no idea. i just stole it from [here](https://github.com/vector-im/element-android/pull/8377/files#diff-8856b3acc5e001dcf87438526083a6636bb2ab86740055ddb438b7741c77d3b1) :P maybe the markdown formatting marks are called 'pills' internally?

Comment (Luka5W) [reply changelog.d/8638.bugfix:1]: @pludi do you have a better message for the changelog or is anyone working on merging on the PR? the code is not reviewed yet

**Review by pludi — APPROVED** [review]

**Comment by pludi** [comment]
Sorry, I didn't know that by just adding a comment I've been set as reviewer…

**Comment by ganfra** [comment]
Replaced by https://github.com/element-hq/element-android/pull/8716

**Pull request closed without merging**